src/main/java/com/autotune/analyzer/services/UpdateRecommendations.java

```diff
@@ -102,111 +90,16 @@ protected void doPost(HttpServletRequest request, HttpServletResponse response)
-                if (null != mainKruizeExperimentMAP.get(experiment_name)) {
-                    kruizeObject = mainKruizeExperimentMAP.get(experiment_name);
-                }
-                if (null != kruizeObject)
-                    experimentResultDataList = new ExperimentDBService().getExperimentResultData(experiment_name, kruizeObject, interval_start_time, interval_end_time);   // Todo this object is not required
-                else {
-                    LOGGER.debug("UpdateRecommendations API request count: {} failed", calCount);
-                    sendErrorResponse(response, null, HttpServletResponse.SC_BAD_REQUEST, String.format("%s%s", MISSING_EXPERIMENT_NAME, experiment_name));
-                    return;
-                }
-            } catch (Exception e) {
-                LOGGER.debug("UpdateRecommendations API request count: {} failed", calCount);
-                sendErrorResponse(response, e, HttpServletResponse.SC_INTERNAL_SERVER_ERROR, e.getMessage());
-                return;
-            }
-
-            if (experimentResultDataList.size() > 0) {
-                //generate recommendation
-                try {
-                    new ExperimentInitiator().generateAndAddRecommendations(kruizeObject, experimentResultDataList, interval_start_time, interval_end_time);    // TODO: experimentResultDataList not required
-                    ValidationOutputData validationOutputData = new ExperimentDBService().addRecommendationToDB(mainKruizeExperimentMAP, experimentResultDataList);
-                    if (validationOutputData.isSuccess()) {
-                        LOGGER.debug("UpdateRecommendations API request count: {} success", calCount);
-                        sendSuccessResponse(response, kruizeObject, interval_end_time);
-                        statusValue = "success";
-                    } else {
-                        LOGGER.debug("UpdateRecommendations API request count: {} failed", calCount);
-                        sendErrorResponse(response, null, HttpServletResponse.SC_INTERNAL_SERVER_ERROR, validationOutputData.getMessage());
-                    }
-                } catch (Exception e) {
-                    LOGGER.debug("UpdateRecommendations API request count: {} failed", calCount);
-                    e.printStackTrace();
-                    LOGGER.error("Failed to create recommendation for experiment: {} and interval_start_time: {} and interval_end_time: {}",
-                            experiment_name,
-                            interval_start_time,
-                            interval_end_time);
-                    sendErrorResponse(response, null, HttpServletResponse.SC_INTERNAL_SERVER_ERROR, e.getMessage());
-                    return;
-                }
+            // validate the input params and generate recommendations
+            RecommendationEngine recommendationEngine = new RecommendationEngine();
```

Comment:
Lets not call empty constructor
initialize RecommendationEngine(experiment_name , interval_endtime)